src/tests/test_max_lhs_option.cpp

```diff
@@ -0,0 +1,60 @@
+#include <filesystem>
+
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/fd/fastfds/fastfds.h"
+#include "algorithms/fd/fd_algorithm.h"
+#include "algorithms/fd/tane/tane.h"
+#include "config/names.h"
+#include "datasets.h"
+#include "model/table/relational_schema.h"
+
+using algos::FastFDs, algos::Tane;
+using config::InputTable;
+
+InputTable GetInputTable(const std::filesystem::path& path, char separator = ',',
+                         bool has_header = true) {
+    return std::make_unique<CSVParser>(path, separator, has_header);
+}
+
+class MaxLHSOptionTest : public ::testing::Test {
+protected:
+    template <typename Algo>
+    static std::unique_ptr<algos::FDAlgorithm> CreateAlgoInstance(unsigned int max_lhs,
+                                                                  bool EqNullsType,
+                                                                  const std::filesystem::path& path,
+                                                                  char separator = ',',
+                                                                  bool hasHeader = true) {
+        using namespace config::names;
+
+        algos::StdParamsMap params{
+                {kCsvPath, path},        {kSeparator, separator},
+                {kHasHeader, hasHeader}, {kTable, GetInputTable(path, separator, hasHeader)},
+                {kMaximumLhs, max_lhs},  {kEqualNulls, EqNullsType}};
+
+        return algos::CreateAndLoadAlgorithm<Algo>(params);
+    }
+};
+
+TEST_F(MaxLHSOptionTest, TaneMaxLHSOptionWork) {
+    unsigned int max_lhs = 3;
+    auto const path = test_data_dir / "TestFD.csv";
```

Comment:
HyFD reports these FDs on this dataset: `[( ) -> 0, ( 5 ) -> 2, ( 5 ) -> 4, ( 5 ) -> 1, ( 5 ) -> 3, ( 2 ) -> 1, ( 2 4 ) -> 5, ( 2 4 ) -> 3, ( 2 3 ) -> 5, ( 2 3 ) -> 4, ( 1 4 ) -> 3, ( 3 4 ) -> 1, ( 1 3 ) -> 4]`. None of them have an LHS longer than 3, so the option does nothing. Either lower the tested value or choose another dataset for these tests.
Also, the 2 tests for Tane and FastFDs look exactly the same. Try `TYPED_TEST_P`, like in `test_algorithm.h`